File: mergesort/mergesort.py
unsorted_list = [27, 369, 63, 126, 252, 990, 54, 18]

def merge_sort(unsorted):
    if len(unsorted) > 1:
        mid = len(unsorted)//2
        left = merge_sort(unsorted[:mid])
        right = merge_sort(unsorted[mid:])  
        result = merge(left, right)
        return result
    else:
        return unsorted

def merge(left, right):
    sorted_list = []
    while len(left) > 0 and len(right) > 0:
        if left[0] <= right[0]:
            sorted_list.append(left.pop(0))
        else:
            sorted_list.append(right.pop(0))
    while len(left) > 0:
        sorted_list.append(left.pop(0))
    while len(right) > 0:
        sorted_list.append(right.pop(0))
    return sorted_list
    
print merge_sort(unsorted_list)

File: mergesort/mergesort.js
const merge_sort = (function(unsorted){
    if(unsorted.length > 1){
        var mid = Math.floor(unsorted.length/2)
        var left = merge_sort(unsorted.slice(0,mid))
        var right = merge_sort(unsorted.slice(mid,))
        return merge(left, right)
    } else {
        return unsorted
    }
})

const merge = (function(left, right){
    var sorted = []
    while(left.length > 0 && right.length > 0){
        left[0] < right [0] ? sorted.push(left.shift()) : sorted.push(right.shift())
    }
    while(left.length>0){
        sorted.push(left.shift())
    }
    while(right.length>0){
        sorted.push(right.shift())
    }
    return sorted
})

console.log(merge_sort([27, 369, 63, 126, 252, 990, 54, 18]))
